src/index.js

```diff
@@ -1,44 +1,71 @@
 /**
- * The author is aware that the following code could be more elegant, especially when I use the library like 
+ * The author is aware that the following code could be more elegant, especially when I use the library like
  * `readline-sync` or `Inquirer.js`.
- * Yet, for simplicity, I've used a very clean and uncomplicated approach based on the variable `mode` and the set of 
+ * Yet, for simplicity, I've used a very clean and uncomplicated approach based on the variable `mode` and the set of
  * constants.
  */
-const inputReader = require('./utils/inputReader')
+const inputReader = require("./utils/inputReader");
+const renderer = require("./utils/renderer");
+const Game = require("./core/game");
+const config = require("./config");
 
-const COMMANDS_MODE = 0, MANOEUVRE_MODE = 1, BATTLE_MODE = 2;
+const COMMANDS_MODE = 0,
+    MANOEUVRE_MODE = 1,
+    BATTLE_MODE = 2;
 
-let mode = COMMANDS_MODE; // the current mode
+let mode = COMMANDS_MODE;
+let game = new Game(config);
 
-while (true)
-{
-    if (mode === COMMANDS_MODE) { // the normal commands mode
-        
+while (!game.isOver) {
+    game.checkIfOver();
```

Comment:
Is it necessary to keep it here? Shouldn't it be at the end of the loop? I would expect the end game check after each move not before it.